src/util/RouterUtil.js

```diff
@@ -218,14 +218,22 @@ function (Okta, OAuth2Util, Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
     case 'FACTOR_REQUIRED':
     case 'FACTOR_CHALLENGE':
     case 'MFA_REQUIRED':
+      if (router.appState.get('hasFailed') && router.appState.get('challengeErrorFactor')) {
```

Comment:
should we change `getDefaultFactor ` to include logic to get the last challenged factor that failed? It might be worth having that logic in the factors instead of here in the router util...

Reply:
Discussed, not doing this